core/domain/skill_domain.py

```diff
@@ -760,36 +744,14 @@ def _update_content_ids_in_assets(self, old_ids_list, new_ids_list):
         written_translations = self.skill_contents.written_translations
         content_ids_to_audio_translations = (
             self.skill_contents.content_ids_to_audio_translations)
-        content_ids_for_text_translations = (
-            written_translations.get_content_ids_for_text_translation())
 
         for content_id in content_ids_to_delete:
-            if not content_id in content_ids_to_audio_translations:
```

Comment:
Any reason why this and the condition below were removed?

Reply:
content_ids_to_audio_translations will necessarily have to contain content_id otherwise we will get a validation error. See:
https://github.com/oppia/oppia/blob/9b5d43d18d8644a9f652788d0b4e07f8d3287fe5/core/domain/skill_domain.py#L318

Reply:
Ok, @DubeySandeep, do you have any thoughts on this?

Reply:
But isn't this independent of that validation test? To activate this, can't you just pass a content_id that doesn't exist in the skill to be deleted? That is what triggers this  condition, right? 
I don't think the line you linked in skill_domain is related to that.

Reply:
The content ids to delete would still remain the same. It is just a set of old content ids - new content ids. Hence any content id that does not exist wont  change the content ids to delete. See here https://github.com/oppia/oppia/blob/9b5d43d18d8644a9f652788d0b4e07f8d3287fe5/core/domain/skill_domain.py#L758

Reply:
I mean, can't you independently call just this function with a manual list of old_ids_list containing IDs not present in the skill to test?

Reply:
Ah I think no. This is a private function which should only be tested through public interfaces